src/Sequence.cc

```diff
@@ -276,15 +310,35 @@ Sequence::ConstIterator Sequence::erase(Sequence::ConstIterator iter)
 {
     throw_if_running();
     auto return_iter = steps_.erase(iter);
+
+    correct_error_index_after_step_erased(return_iter);
     enforce_invariants();
     return return_iter;
 }
 
 Sequence::ConstIterator Sequence::erase(Sequence::ConstIterator begin,
                                         Sequence::ConstIterator end)
 {
+    const auto erased_begin_idx = std::distance(cbegin(), begin);
+    const auto erased_end_idx = std::distance(cbegin(), end);
```

Comment:
Do we have the precondition that `begin <= end`?

According to [C++ reference](https://en.cppreference.com/w/cpp/iterator/distance) it can also be inverted. I don't know if it has any effect of the below conditions...

Reply:
Yes, we have that precondition. But we can affort to check that and throw an exception, so maybe we should. I'll add a check.

Reply:
Added a check.